src/functional/otf/binding/pybind.py

```diff
@@ -108,11 +110,15 @@ class BindingCodeGenerator(TemplatedGenerator):
     )
 
     def visit_FunctionParameter(self, param: FunctionParameter):
-        if param.ndim > 0:
+        if isinstance(param.type_, ts.FieldType):
             type_str = "pybind11::buffer"
+        elif isinstance(param.type_, ts.IndexFieldType):
+            type_str = "pybind11::none"
```

Comment:
There is no object shared across C++ and Python that could represents index fields. Furthermore, index fields are currently fully described by their type, they contain have no data members.